crud-operations/src/main/java/org/example/crudoperations/service/ProductService.java

```diff
@@ -0,0 +1,50 @@
+package org.example.crudoperations.service;
+
+import org.example.crudoperations.entity.Product;
+import org.example.crudoperations.repository.ProductRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class ProductService {
+    @Autowired
+    private ProductRepository repository;
+
+    public Product saveProduct(Product product){
+       return repository.save(product);
+    }
+
+    public List<Product> saveProducts(List<Product> products){
+        return repository.saveAll(products);
+    }
+    public List<Product> getProducts(){
+       return repository.findAll();
+
+    }
+    public Product getProductById(UUID id){
+        return repository.findById(id).orElse(null);
+    }
+
+    public void deleteProduct(UUID id){
+         repository.deleteById(id);
+    }
+
+    public Product updateProduct(UUID id,Product product){
+        Optional<Product> productToUpdateOptional = repository.findById(id);
```

Comment:
We don't have to fetch the Product only to update it. Directly update the details and save it to make it faster.

Reply:
i m fetching product from db only to check if it exist or not as id coming from frontend can be wrong as well

Reply:
JPA throws exception if the id is not present which can be handled here. You can play around with it to understand more.